web-ng/src/app/services/auth/auth.guard.ts

```diff
@@ -14,36 +14,55 @@
  * limitations under the License.
  */
 import { Injectable } from '@angular/core';
-import { Router } from '@angular/router';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+} from '@angular/router';
 import { AuthService } from './auth.service';
 import { Observable, of } from 'rxjs';
 import { catchError, map } from 'rxjs/operators';
 import { environment } from '../../../environments/environment';
+import { User } from '../../shared/models/user.model';
 
 @Injectable({
   providedIn: 'root',
 })
-export class AuthGuard {
-  private static readonly SIGN_IN_URL = '/signin';
-
+export class AuthGuard implements CanActivate {
   constructor(private router: Router, private authService: AuthService) {}
 
-  canActivate(): Observable<boolean> {
+  canActivate(
+    route: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot
+  ): Observable<boolean> {
     return this.authService.getUser$().pipe(
-      map(response => {
-        if (environment.useEmulators) {
-          return true;
+      map(user => {
+        if (state.url === '/signin') {
```

Comment:
Will `url` also return any URL fragment (#anchor) or query (?a=123) params? If so this won't work if the user adds them to the URL. Not sure, but I suspect you may need to use `ActivatedRouteSnapshot` instead.

Reply:
I am using string includes which will check for substring now.